runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -921,20 +925,458 @@ static void VMwrtbarEvaluator(TR::Node *node, TR::Register *srcReg, TR::Register
+   // Load value from dataSnippetRegister + offsetInDataBlock then compare and branch
+   generateTrg1MemInstruction(cg, TR::InstOpCode::Op_load, node, scratchReg,
+                     new (cg->trHeapMemory()) TR::MemoryReference(dataSnippetRegister, offsetInDataBlock, TR::Compiler->om.sizeofReferenceAddress(), cg));
+   generateTrg1Src1ImmInstruction(cg, TR::InstOpCode::andi_r, node, scratchReg, scratchReg, cndReg, -1);
+   generateConditionalBranchInstruction(cg, TR::InstOpCode::bne, node, unresolvedLabel, cndReg);
+   
+   generateReportOOL->swapInstructionListsWithCompilation();
+      {
+      generateLabelInstruction(cg, TR::InstOpCode::label, node, unresolvedLabel);
+
+      if (isStatic)
+         {
+         // Fill in J9JITWatchedStaticFieldData.fieldClass
+         TR::Register *fieldClassReg = NULL;
+
+         if (isWrite)
+            {
+            fieldClassReg = cg->allocateRegister();
+            generateTrg1MemInstruction(cg,TR::InstOpCode::Op_load, node, fieldClassReg,
+                           new (cg->trHeapMemory()) TR::MemoryReference(sideEffectRegister, comp->fej9()->getOffsetOfClassFromJavaLangClassField(), TR::Compiler->om.sizeofReferenceAddress(), cg));
+            }
+         else
+            {
+            fieldClassReg = sideEffectRegister;
+            }
+
+         // Store value to dataSnippetRegister + offset of fieldClass
+         if (is64Bit)
+            {
+            TR::MemoryReference *memRef = new (cg->trHeapMemory()) TR::MemoryReference(dataSnippetRegister, offsetof(J9JITWatchedStaticFieldData, fieldClass), TR::Compiler->om.sizeofReferenceAddress(), cg);
+            generateMemSrc1Instruction(cg, TR::InstOpCode::std, node, memRef, fieldClassReg);
+            }
+         else 
+            {
+            generateMemSrc1Instruction(cg, TR::InstOpCode::stw, node, new (cg->trHeapMemory()) TR::MemoryReference(dataSnippetRegister, offsetof(J9JITWatchedStaticFieldData, fieldClass), 4, cg), fieldClassReg->getLowOrder());
+            generateMemSrc1Instruction(cg, TR::InstOpCode::stw, node, new (cg->trHeapMemory()) TR::MemoryReference(dataSnippetRegister, offsetof(J9JITWatchedStaticFieldData, fieldClass) + 4, 4, cg), fieldClassReg->getHighOrder());
+            }
+         deps->addPreCondition(fieldClassReg, TR::RealRegister::NoReg);
+         deps->addPostCondition(fieldClassReg, TR::RealRegister::NoReg);
+         }
+
+      TR::ResolvedMethodSymbol *methodSymbol = node->getByteCodeInfo().getCallerIndex() == -1 ? comp->getMethodSymbol(): comp->getInlinedResolvedMethodSymbol(node->getByteCodeInfo().getCallerIndex());      
+
+      // Store cpAddressReg
+      // TODO: Replace when AOT TR_ConstantPool Discontiguous support is enabled
+      //loadAddressConstant(cg, node, (uintptrj_t)methodSymbol->getResolvedMethod()->constantPool(), resultReg, NULL, false, TR_ConstantPool);
```

Comment:
Can we open an issue to track this work?

Reply:
Referenced this in [1].

[1] https://github.com/eclipse/openj9/issues/6256#issuecomment-505652613